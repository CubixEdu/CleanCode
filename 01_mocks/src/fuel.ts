export class Fuel {

    public fill(litersOfFuel: number) {
        return litersOfFuel;
    }
}