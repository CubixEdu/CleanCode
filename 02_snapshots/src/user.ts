export class User {
  
    constructor(
      private id: number,
      private username: string,
      private email: string,
      private firstName: string,
      private lastName: string,
      private dateOfBirth: Date,
      private address: {
        street: string;
        city: string;
        zipCode: string;
      },
      private phoneNumber: string,
      private isActive: boolean,
      private roles: string[]
    ) {}
  }