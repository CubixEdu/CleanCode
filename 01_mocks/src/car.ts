import { Fuel } from "./fuel";

export class Car {

    constructor(private fuel: Fuel) { }

    public drive() {
        this.fuel.fill(35);
    }
}