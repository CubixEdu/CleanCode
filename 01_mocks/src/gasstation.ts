import { Fuel } from "./fuel";

export class GasStation {

    constructor(private fuel: Fuel) { }

    public fillUpCars(numberOfCars: number) {
        for (let index = 1; index < numberOfCars + 1; index++) {
            this.fuel.fill(index + 1);
        }
    }
}