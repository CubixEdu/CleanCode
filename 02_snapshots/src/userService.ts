import { User } from "./user";

export class UserService {

  public getUserById(id: number) {
    // simulate a real user data retrieval here
    return new User(
      id,
      'john_doe',
      'john@example.com',
      'John',
      'Doe',
      new Date('1990-01-15'),
      {
        street: '123 Main St',
        city: 'Anytown',
        zipCode: '12345',
      },
      '123-456-7890',
      true,
      ['user', 'admin']
    );
  }
}
