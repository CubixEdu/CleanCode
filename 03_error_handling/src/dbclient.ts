import { User } from "./user";

export class DbClient {

    public getUserByUsername(username: string) {
        return new User(
            1,
            'john_doe',
            'john@example.com',
            'John',
            'Doe',
            new Date('1990-01-15')
          );
    }
}