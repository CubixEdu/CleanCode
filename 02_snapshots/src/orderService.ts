import { UserService } from "./userService";

export class OrderService {

    constructor(private userService: UserService) { }

    public getUsersOfTheLastHour() {
        const users = [];
        for (let index = 1; index < 5; index++) {
            const user = this.userService.getUserById(index);
            users.push(user);
        }
        return users;
    }
}