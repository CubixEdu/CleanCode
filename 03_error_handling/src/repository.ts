import { NetworkError } from "./exceptions/NetworkError";
import { NotFoundError } from "./exceptions/NotFoundError";
import { UnknownError } from "./exceptions/UnknownError";
import { ValidationError } from "./exceptions/ValidationError";
import { DbClient } from "./dbclient";
import { Logger } from "./logger";

export class Repository {

    constructor(private dbClient: DbClient, private logger: Logger) { }

    public getUserByUsername(username: string) {
        this.validateInput(username);

        try {
            return this.dbClient.getUserByUsername(username);
        } catch (error) {
            if (error instanceof NetworkError || error instanceof NotFoundError) {
                this.logger.logError(error.message, error);
                throw error;
            }

            throw new UnknownError('Unknown error happened.');
        }
    }

    private validateInput(username: string) {
        if (!username) {
            throw new ValidationError('username cannot be null or empty.');
        }
    }
}