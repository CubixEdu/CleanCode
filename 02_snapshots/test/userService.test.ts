import { UserService } from "../src/userService";


describe("UserService tests", () => {
   
    test('Given I have an existing user, When I want to retrive it by ID, Then the user with all properties is returned', () => {
        // Arrange
        const userService = new UserService();

        // Act
        const actualResult = userService.getUserById(1);

        // Assert
        expect(actualResult).toMatchSnapshot();
    })
})