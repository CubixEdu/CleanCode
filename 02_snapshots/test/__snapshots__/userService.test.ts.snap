// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserService tests Given I have an existing user, When I want to retrive it by ID, Then the user with all properties is returned 1`] = `
User {
  "address": {
    "city": "Anytown",
    "street": "123 Main St",
    "zipCode": "12345",
  },
  "dateOfBirth": 1990-01-15T00:00:00.000Z,
  "email": "john@example.com",
  "firstName": "John",
  "id": 1,
  "isActive": true,
  "lastName": "Doe",
  "phoneNumber": "123-456-7890",
  "roles": [
    "user",
    "admin",
  ],
  "username": "john_doe",
}
`;
